src/pudl/extract/extractor.py

```diff
@@ -333,27 +356,46 @@ def years_from_settings_factory(name: str) -> OpDefinition:
     @op(
         out=DynamicOut(),
         required_resource_keys={"dataset_settings"},
-        name=f"{name}_years_from_settings",
+        name=f"{name}_partitions_from_settings",
     )
-    def years_from_settings(context) -> DynamicOutput:
-        """Produce target years for the given dataset from the dataset settings object.
+    def partitions_from_settings(context) -> DynamicOutput:
+        """Produce target partitions for the given dataset from the dataset settings.
 
         These will be used to kick off worker processes to extract each year of data in
         parallel.
 
         Yields:
-            A Dagster :class:`DynamicOutput` object representing the year to be
+            A Dagster :class:`DynamicOutput` object representing the partition to be
             extracted. See the Dagster API documentation for more details:
             https://docs.dagster.io/_apidocs/dynamic#dagster.DynamicOut
         """
         if "eia" in name:  # Account for nested settings if EIA
-            year_settings = context.resources.dataset_settings.eia
+            partition_settings = context.resources.dataset_settings.eia
         else:
-            year_settings = context.resources.dataset_settings
-        for year in getattr(year_settings, name).years:
-            yield DynamicOutput(year, mapping_key=str(year))
+            partition_settings = context.resources.dataset_settings
+        # Get year/year_quarter/half_year partition
+        data_settings = getattr(partition_settings, name)  # Get dataset settings
+
+        partition = [
+            var
+            for var in vars(data_settings)
+            if any(
+                date_partition in var
+                for date_partition in ["years", "half_years", "year_quarters"]
+            )
+        ]
+        assert (
+            len(partition) == 1
+        ), f"Only one working partition is supported: {partition}."
```

Comment:
just for my clarification: this is so we don't try to get "years" and also "half_years"?